### PR TITLE
Annotate exceptions in packages / standard / Structured data

### PR DESCRIPTION
**Pull request by Demetri0**
I hope that I correctly understand the task (#154), if I done something wrong please let me know, and I will try to fix it.

It is a bit crazy to make commit for each line, but I want to provide reference where I take exceptions list. I think it would be better to squash it.

### CODE REVIEW

**Comment by JSMonk** [comment]
@Demetri0, Yes, only one thing. Hegel tries to prevent any `TypeError`, so could you please drop any `TypeError`, because after Hegel analysis it should not be presented.
Thank you for your contribution ^_^.

**Comment by JSMonk** [comment]
@Demetri0 
P.S. Only if it's not possible to analyze the case. For Example, Currently, we don't detect circular dependency inside objects, so, `JSON.stringify` can throw a `TypeError`

**Comment by Demetri0** [comment]
I'll split for several PR's

File: packages/typings/standard/index.d.ts [hunk 1]
```diff
@@ -1870,7 +1872,8 @@ interface ArrayBufferView {
 
 interface ArrayBufferConstructor {
   readonly prototype: ArrayBuffer;
-  new (byteLength: number): ArrayBuffer;
+  // @throws {RangeError} in case byteLength > 2**32 - 1
+  new (byteLength: number): ArrayBuffer | $Throws<RangeError>;
```
Comment (JSMonk) [inline packages/typings/standard/index.d.ts:1876]: `RangeError` or `TypeError`?

Comment (Demetri0) [reply packages/typings/standard/index.d.ts:1876]: Screenshot from 2020-04-25 20-30-56

**Pull request merged**